src/librustc_mir/interpret/place.rs

```diff
@@ -740,7 +740,7 @@ where
         // but not factored as a separate function.
         let mplace = match dest.place {
             Place::Local { frame, local } => {
-                match self.stack_mut()[frame].locals[local].access_mut()? {
+                match M::access_local_mut(self, frame, local)? {
```

Comment:
There's a risk here of accidentally using the frame method instead of the hook. Is there anything we can do about that?

I think there should at least be a comment at the frame method saying to not call it (except when implementing the hook).

Reply:
I'll investigate whether we can have something static and otherwise document it properly

Reply:
I documented it. I don't think it's realistic to statically prevent it being called except by introducing some appropriately named token types.